scrapy/pipelines/images.py

```diff
@@ -114,25 +114,25 @@ def file_downloaded(self, response, request, info, *, item=None):
         return self.image_downloaded(response, request, info, item=item)
 
     def image_downloaded(self, response, request, info, *, item=None):
-        
+        logger = logging.getLogger()
         checksum = None
         image_stream = self.get_images(response, request, info, item=item)
         while True:
             try:
                 path, image, buf = next(image_stream)
-            except OSError as e:
+            except OSError:
                 logger.exception('Could not process image')
```

Comment:
There should be a `continue` in this case, otherwise I imagine we would end up with duplicate processing of the image from the previous iteration.